Makefile

```diff
@@ -13,4 +13,149 @@ collector: libkindling
 .PHONY: libkindling
 libkindling:
 	mkdir -p probe/build && cd probe/build && cmake -DBUILD_DRIVER=OFF -DPROBE_VERSION=0.1.1dev .. && make 
-	mkdir -p collector/docker/libso/. && cp -rf probe/build/src/libkindling.so collector/docker/libso/. && cp -rf probe/build/src/libkindling.so /usr/lib64/.
\ No newline at end of file
+	mkdir -p collector/docker/libso/. && cp -rf probe/build/src/libkindling.so collector/docker/libso/. && cp -rf probe/build/src/libkindling.so /usr/lib64/.
+
+
+
+LIBS_SRC := $(abspath ./probe/libs/agent-libs)
+COLLECTOR_PATH := $(abspath ./collector)
+SCRIPTS_PATH := $(abspath ./deploy/scripts)
+PROBE_PATH := $(abspath ./probe)
+PROBE_PATCH_FILE := $(PROBE_PATH)/cmake/modules/agent-libs.cmake
+
+kernelVersion = $(shell uname -r)
+ErrorExit = exit 1
+
+ifneq ($(shell cat /etc/os-release | grep Ubuntu | wc -l),0)
+	KernelName = linux-headers-$(kernelVersion)
+	CheckCommand = dpkg -l | grep $(KernelName)
+	InstallCommand = sudo apt-get -y install $(KernelName)
+else
+	ifneq ($(shell cat /etc/os-release | grep CentOS | wc -l),0)
+		KernelName = kernel-devel-$(kernelVersion)
+		CheckCommand = yum list installed |grep $(KernelName)
+		InstallCommand = sudo yum -y install $(KernelName)
+	else
+		KernelName = NotSupport
+		CheckCommand = $(ErrorExit)
+		InstallCommand = $(ErrorExit)
+	endif
+endif
```

Comment:
Looks like these are only needed when running `make kernel-headers`. Why not move these under that part?

By the way, I figured `CheckCommand` is never used and should be removed.

Reply:
Moved to `make kernel-headers`.